### PR TITLE
[F3D] Pipe Sync Optimization

### PR DESCRIPTION
**Pull request by jesusyoshi54**
gsSP cmds will export before using a pipe sync, which will theoretically save a minor amount of time. Mat bleed will also optimize pipe syncs within a mesh by removing pipe sync cmds if there is no gsDP cmd.

### CODE REVIEW

File: fast64_internal/f3d/f3d_writer.py [hunk 1]
```diff
@@ -1313,12 +1313,26 @@ def saveOrGetF3DMaterial(material, fModel, obj, drawLayer, convertTextureData):
...
+
+    defaults = bpy.context.scene.world.rdp_defaults
+    if fModel.f3d.F3DEX_GBI_2:
+        saveGeoModeDefinitionF3DEX2(fMaterial, f3dMat.rdp_settings, defaults, fModel.matWriteMethod)
+    else:
+        saveGeoModeDefinition(fMaterial, f3dMat.rdp_settings, defaults, fModel.matWriteMethod)
+
+    # Checking for f3dMat.rdp_settings.g_lighting here will prevent accidental exports,
+    # There may be some edge case where this isn't desired.
+    if useDict["Shade"] and f3dMat.set_lights and f3dMat.rdp_settings.g_lighting:
```
Comment (Lilaa3) [inline fast64_internal/f3d/f3d_writer.py:1330]: extreme nitpick but id change this to
```suggestion
    if useDict["Shade"] and f3dMat.rdp_settings.g_lighting and f3dMat.set_lights:
```

**Review by Lilaa3 — COMMENTED** [review]
The logic looks to be perfect aside from the thing I commented on, my export of mario worked perfectly ingame and the displaylist looked just as perfect. This is a great addition!

File: fast64_internal/f3d/f3d_writer.py (hunk 1 above)
Comment (jesusyoshi54) [reply fast64_internal/f3d/f3d_writer.py]: Sure 👍

**Review by Lilaa3 — APPROVED** [review]
Ship it

**Pull request merged**